src/components/App.tsx

```diff
@@ -2892,6 +2892,14 @@ class App extends React.Component<ExcalidrawProps, AppState> {
         pointerDownState.origin.y,
         elementType === "draw" ? null : this.state.gridSize,
       );
+
+      // If arrow is pre-decorators, it will have undefined for both start and end decorators.
+      // If so, we want it to be null/"arrow". If the linear item is not an arrow, we want it
+      // to be null/null. Otherwise, we want it to use the currentItemLinearDecorators values.
+      const { start, end } = this.state.currentItemLinearDecorators;
+      const [startDecorator = null, endDecorator = "arrow"] =
+        elementType === "arrow" ? [start, end] : [null, null];
+
```

Comment:
I believe this was to catch arrows being copy and pasted between an older pre-decorators version and PR's version. I suppose this would never occur in production, though, so I'll remove this bit.